22-Drawing App/script.js

```diff
@@ -0,0 +1,85 @@
+const canvas = document.getElementById('canvas')
+const increaseBtn = document.getElementById('increase')
+const decreaseBtn = document.getElementById('decrease')
+const sizeEl = document.getElementById('size')
+const colorEl = document.getElementById('color')
+const clearEl = document.getElementById('clear')
+
+const ctx = canvas.getContext('2d')
+
+let size = 10
+isPressed = false
+let color = 'black'
+let x
+let y
+
+canvas.addEventListener('mousedown', (e) => {
+  isPressed = true
+
+  x = e.offsetX
+  y = e.offsetY
+})
+
+document.addEventListener('mouseup', (e) => {
+  isPressed = false
+
+  x = undefined
+  y = undefined
+})
+
+canvas.addEventListener('mousemove', (e) => {
+  if (isPressed) {
+    const x2 = e.offsetX
+    const y2 = e.offsetY
+
+    drawCircle(x2, y2)
+    drawLine(x, y, x2, y2)
+
+    x = x2
+    y = y2
+  }
+})
```

Comment:
indention wise, this function could be much cleaner if you invert the case!
ie. instead of checking to see `if (isPressed) {...}`, and then you have to indent pretty much all of the function content, you could go:
```
if (!isPressed) return;

const x2 = e.offsetX
const y2 = e.offsetY

drawCircle(x2, y2)
drawLine(x, y, x2, y2)
x = x2
y = y2
```